app/domain/services/fetch_course_events/service.rb

```diff
@@ -353,9 +354,10 @@ def fetch_and_process_course_events(courses, event_types, restart, current_time)
             is_pe: assigned_exercise.fetch(:is_pe)
           )
 
-          no_feedback_assigned_exercise_uuids << assigned_exercise_uuid \
-            if (!due_at.nil? && due_at > current_time) ||
-               (!feedback_at.nil? && feedback_at > current_time)
+          no_feedback_assigned_exercise_uuids << assigned_exercise_uuid if !is_deleted && (
+            (!due_at.nil? && due_at > current_time) ||
+            (!feedback_at.nil? && feedback_at > current_time)
+          )
         end
```

Comment:
nitpick: to me, `due_at.present?` reads clearer than `!due_at.nil?`

Reply:
I agree.